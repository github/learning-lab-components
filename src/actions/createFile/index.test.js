const createFile = require('./index')
const mockContext = require('../../tests/mockContext')

describe('createFile', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      pull_request: {
        head: {
          sha: 123
        }
      }
    }, {
      repos: {
        createFile: jest.fn()
      }
    })
  })

  it('creates a new file', async () => {
    context.fromFile = () => Promise.resolve('Hello!')
    await createFile(context, { filename: 'filename' })
    expect(context.github.repos.createFile.mock.calls).toMatchSnapshot()
  })

  it('replaces file contents with context variables', async () => {
    await createFile(context, {
      filename: 'filename',
      content: 'Hello {{ login }}!',
      branch: 'a-feature-branch',
      data: {
        login: 'Jason'
      }
    })

    expect(context.github.repos.createFile.mock.calls).toMatchSnapshot()
  })

  it('can choose a new filename', async () => {
    await createFile(context, { filename: 'filename', new_name: 'CODEOWNERS' })
    expect(context.github.repos.createFile.mock.calls).toMatchSnapshot()
  })

  it('can set a custom commit message', async () => {
    await createFile(context, { filename: 'filename', message: 'Hello!' })
    expect(context.github.repos.createFile.mock.calls[0][0].message).toBe('Hello!')
  })

  it('creates a new file on the given branch', async () => {
    context.fromFile = () => Promise.resolve('Hello!')
    await createFile(context, { filename: 'filename', branch: 'a-feature-branch' })
    expect(context.github.repos.createFile.mock.calls).toMatchSnapshot()
  })
})
