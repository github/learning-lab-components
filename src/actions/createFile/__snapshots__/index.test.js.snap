// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createFile can choose a new filename 1`] = `
Array [
  Array [
    Object {
      "content": "ZmlsZW5hbWU=",
      "message": "Creating new file CODEOWNERS",
      "owner": "JasonEtco",
      "path": "CODEOWNERS",
      "repo": "example",
    },
  ],
]
`;

exports[`createFile creates a new file 1`] = `
Array [
  Array [
    Object {
      "content": "SGVsbG8h",
      "message": "Creating new file filename",
      "owner": "JasonEtco",
      "path": "filename",
      "repo": "example",
    },
  ],
]
`;

exports[`createFile creates a new file on the given branch 1`] = `
Array [
  Array [
    Object {
      "branch": "a-feature-branch",
      "content": "SGVsbG8h",
      "message": "Creating new file filename",
      "owner": "JasonEtco",
      "path": "filename",
      "repo": "example",
    },
  ],
]
`;

exports[`createFile replaces file contents with context variables 1`] = `
Array [
  Array [
    Object {
      "branch": "a-feature-branch",
      "content": "ZmlsZW5hbWU=",
      "message": "Creating new file filename",
      "owner": "JasonEtco",
      "path": "filename",
      "repo": "example",
    },
  ],
]
`;
